eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -566,7 +567,13 @@ object Router {
 
   case class ShortChannelIdAndFlag(shortChannelId: ShortChannelId, flag: Long)
 
-  case class Syncing(pending: List[RoutingMessage], total: Int)
+  /**
+   * @param remainingQueries remaining queries to send, the next one will be popped after we receive a [[ReplyShortChannelIdsEnd]]
+   * @param totalQueries     total number of *queries* (not channels) that will be sent during this syncing session
+   */
+  case class Syncing(remainingQueries: List[RoutingMessage], totalQueries: Int) {
+    def started: Boolean = totalQueries > 0
```

Comment:
nit: I read the comments about `val` vs `def` here but even when we call `.copy()` on a case class its constructors will be called and a `val` would be safe

Reply:
Interesting, let's keep it like this though where it's obvious that it will work?